### ISSUE TITLE
Discord OAuth and User Accounts

### ISSUE THREAD

**Issue by TheScree** [MalekaiProject/crowfallWiki#4]
Integrate our initial and primary OAuth login source, Discord. The only other login system I'd be okay at looking at for account-creation on the app ecosystem is Twitter. Trying to keep the various logins to things most people would be okay with sharing in a game environment. Steam is apparently garbage at OAuth so we'll skip that for now....

- [ ] Discord OAuth Based Account Creation
- [ ] Non-OAuth Based Account Creation
- [ ] Swap Issue Flagging to only be do-able by users with accounts.

**Comment by MichaelFedora** [comment]
add github oauth as well imo. CF needs to get  oauth implemented...  that'd be sweet, but oh well.

Otherwise this needs to be implemented `malekai.network` side, not cf wiki. Regardless, issue flagging only by users should be swapped over, yeah.

Lastly, should we stick to oauth only if we can(?). I don't want to store passwords if we don't have to.

**Comment by TheScree** [comment]
I agree about not storing passwords. I'm okay with requiring an oauth route to verify people. Makes user management super easy, and allows people to trust us more, considering we won't be storing passwords (any fansite that does this forces players to make up brand new passwords to avoid people running the site trying to login as them).

Github, Discord and _maybe_ Steam? Only three I'd worry about.
